internal/transform/schema.go

```diff
@@ -30,25 +32,30 @@ type LedgerOutput struct {
 
 // TransactionOutput is a representation of a transaction that aligns with the BigQuery table history_transactions
 type TransactionOutput struct {
-	TransactionHash      string    `json:"transaction_hash"`
-	LedgerSequence       uint32    `json:"ledger_sequence"`
-	ApplicationOrder     uint32    `json:"application_order"`
-	Account              string    `json:"account"`
-	AccountMuxed         string    `json:"account_muxed,omitempty"`
-	AccountSequence      int64     `json:"account_sequence"`
-	MaxFee               uint32    `json:"max_fee"`
-	FeeCharged           int64     `json:"fee_charged"`
-	OperationCount       int32     `json:"operation_count"`
-	CreatedAt            time.Time `json:"created_at"`
-	MemoType             string    `json:"memo_type"`
-	Memo                 string    `json:"memo"`
-	TimeBounds           string    `json:"time_bounds"`
-	Successful           bool      `json:"successful"`
-	TransactionID        int64     `json:"id"`
-	FeeAccount           string    `json:"fee_account,omitempty"`
-	FeeAccountMuxed      string    `json:"fee_account_muxed,omitempty"`
-	InnerTransactionHash string    `json:"inner_transaction_hash,omitempty"`
-	NewMaxFee            uint32    `json:"new_max_fee,omitempty"`
+	TransactionHash             string         `json:"transaction_hash"`
+	LedgerSequence              uint32         `json:"ledger_sequence"`
+	ApplicationOrder            uint32         `json:"application_order"`
+	Account                     string         `json:"account"`
+	AccountMuxed                string         `json:"account_muxed,omitempty"`
+	AccountSequence             int64          `json:"account_sequence"`
+	MaxFee                      uint32         `json:"max_fee"`
+	FeeCharged                  int64          `json:"fee_charged"`
+	OperationCount              int32          `json:"operation_count"`
+	CreatedAt                   time.Time      `json:"created_at"`
+	MemoType                    string         `json:"memo_type"`
+	Memo                        string         `json:"memo"`
+	TimeBounds                  string         `json:"time_bounds"`
+	Successful                  bool           `json:"successful"`
+	TransactionID               int64          `json:"id"`
+	FeeAccount                  string         `json:"fee_account,omitempty"`
+	FeeAccountMuxed             string         `json:"fee_account_muxed,omitempty"`
+	InnerTransactionHash        string         `json:"inner_transaction_hash,omitempty"`
+	NewMaxFee                   uint32         `json:"new_max_fee,omitempty"`
+	LedgerBounds                string         `json:"ledger_bounds"`
+	MinAccountSequence          null.Int       `json:"min_account_sequence"`
+	MinAccountSequenceAge       null.Int       `json:"min_account_sequence_age"`
```

Comment:
in the[ xdr model on new p19 preconditions](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0021.md#specification), `MinAccountSequence` was nullable, but the other two(SequenceAge/LedgerGap) were non-nullable, seems like could be safely typed a bit more strongly here to same, but maybe this is already more into the extraction model now, just wanted to mention.

Reply:
@sreuland do you know why [Horizon](https://github.com/stellar/go/blob/6af10a1f8a3f6d6d5775d8d2718779361f3c1b6d/services/horizon/internal/db2/history/transaction_batch_insert_builder.go#L136-L138) defines these preconditions as `null.Int`? For most data types in Hubble, I try and mirror whatever is implemented in Horizon DB. (Maybe @Shaptic knows?)

I think your hypothesis is correct, I just want to confirm.

Reply:
I see, followed through horizon references a bit on these attributes, and yes, they use nullable references, probably because prior to p19, tx's wouldn't have presence of that yet, the usage here is consistent, sorry for the detour, looks fine!